File: Old C Projects/Hanoi/LinkedList.cpp
#include "LinkedList.h"
#include <string>
#include <iostream>
using namespace std;

LinkedList::LinkedList ()
{
	head = NULL;
	ct = 0;
}

LinkedList::~LinkedList ()
{
	cout << "Deleting the Linked List!" << endl;
	delete head;
}

int LinkedList::pop()
{
	int val;
	if (head == NULL){return 0;}
	else 
	{
		val = head->getnum();
		head = head->pop();
	}
	return val;
}

void LinkedList::push (int N) 
{
	if (head==NULL){head = new LLN (N,NULL);}
	else 
	{
		LLN *p = new LLN (N,head);
		head = p;
	}
	ct++;
}

void LinkedList::print()
{
	if (head == NULL)
		cout << "There is nothing to print!" << endl;
	else 
	{
		head->print();
	}
}

	

File: Old Java Projects/LinkedLists/LinkedList.java
public class LinkedList {

 private LinkedListNode head = null;

 public LinkedListNode gethead () {return head;}

 public void Add (String s, double t)
 	{
	 if (head != null) head.Add(s,t);
	 else head = new LinkedListNode (s,t, null);
	}

 public void print () //Prints the nothing if its empty, goes to LLN if its not.
 {
	 if (head != null) head.print();
	 System.out.println ();
 }

 public void Delete (String s) //Returns an error if there is no one to delete; else it goes to LLN delete.
 {
	 if (head != null) head = head.Delete (s);
	 else System.out.println ("ERROR: THERE IS NOTHING TO DELETE!");
 }

 public void Search (String s)// Returns nothing if its empty, else goes to LLN to search.
 {
	 if (head != null) head.Search(s);
	 if (head == null) System.out.println (s+" wasn't found but would be at the beginning of the list.");
	 System.out.println();
}
}